ports/atomvm_mqtt_client.c

```diff
@@ -119,8 +119,7 @@ struct platform_data {
 
 static term make_atom(GlobalContext *global, const char *string)
 {
-    int global_atom_index = globalcontext_insert_atom(global, (AtomString) string);
-    return term_from_atom_index(global_atom_index);
+    return globalcontext_make_atom(global, string);
```

Comment:
The results of globalcontext_make_atom() should be checked to be sure its a valid atom:
```c
static term make_atom(GlobalContext *global, const char *string)
{
    term new_atom = globalcontext_make_atom(global, string);
    if (UNLIKELY(term_is_invalid_term(new_atom))) {
        RAISE_ERROR(OUT_OF_MEMORY_ATOM);
    }
    
    return new_atom
}
```
See atomvm/AtomVM#1442

Reply:
thanks, learn something new every day..